journal/src/test/java/io/camunda/zeebe/journal/file/SegmentedJournalTest.java

```diff
@@ -624,6 +624,27 @@ void shouldBeAbleToResetAgainWhileThePreviousFileIsNotDeleted() {
         .hasSize(1);
   }
 
+  @Test
+  void shouldReleaseReadLock() {
+    // given
+    final var journal = openJournal(5);
+
+    // delete first segment so that it closes
+    final var segment = journal.getFirstSegment();
+    segment.delete();
+
+    // expect
+    assertThat(segment.isOpen()).isFalse();
+    assertThatThrownBy(() -> journal.openReader()).withFailMessage("Segment not open");
+
+    // when
+    journal.reset(100);
+
+    // then
+    assertThat(journal.getJournalIndex().lookup(journalIndexDensity)).isNull();
+    assertThat(journal.getJournalIndex().lookup(2 * journalIndexDensity)).isNull();
```

Comment:
:thought_balloon: I don't understand how these two assertions test that the read lock is released. I think what we are really testing here is whether the `journal.reset(...)` finishes?

Reply:
Correct! The test case tests whether the read lock is released implicitly when `journal.reset(...)` finishes. Because it is not possible to assert the release of the read lock explicitly, I decided to assert the effects that result out of `journal.reset()` (which I copied and pasted from another test case in the same test class). Otherwise, a test case without any assertions at the end does not seem to be useful (at least for me).

What could we do to improve that test case?

1. Add a comment to the assertions to make my intent transparent/explicit.
2. Assert other things, but not sure which one.
3. Rewrite the test case may be as follows (pseudo-code):
```
@Test
void shouldReleaseReadLock() {
  // given
  final var journal = openJournal(5);

  // delete first segment so that it closes
  final var segment = journal.getFirstSegment();
  segment.delete();

  // expect
  assertThat(segment.isOpen()).isFalse();
  assertThatThrownBy(() -> journal.openReader()).withFailMessage("Segment not open");

  final var latch = new CountDownLatch(1)

  // when
 new Thread(() -> {
    journal.reset(100));
    latch.countDown();
  }).start();

  assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
}
```

After giving it a second thought, I would go with the third option (i.e., re-writing the test case as suggested). That way, the test case is much more explicit about the scenario and the expected result.

WDYT? Do you see any other alternatives?

Reply:
I like option three, let's do it this way :+1:

Reply:
What I not like is that we couple these test a lot how it is implemented, which is likely to change and than we need to change this test. But I see currently no other better way unfortunately, except moving the lock/release logic into a separate class and using it as a composite in the journal. In that way we could test the lock/release logic separately.

Reply:
Thanks for your input, @Zelldon. I think I get your point. However, to keep this PR as simple as possible, I would suggest including your suggestion as a to-do in the separate issue mentioned in the comment below: https://github.com/camunda-cloud/zeebe/pull/8372#discussion_r768483240. Meaning, when checking the code-base for other locking patterns, it turns out that there are several occurrences of such a locking pattern, then we should put that concern in a separate class and adjust the test cases as necessary. But for now, I would go with the test case as suggested above.

Are you okay with that approach?

Reply:
Yes thanks @romansmirnov